### PR TITLE
init

### PR DESCRIPTION
**Pull request by sbs2001**

### CODE REVIEW

File: src/crowdsec.ts [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+    
+    const outcome =  await result.json();
+    return outcome["success"] === true;
+}
+
+
+async function serveCaptchaSubmissionHandler(r: NginxHTTPRequest): Promise<void> {
+    if(await captchaSubmissionIsCorrect(r) == true){
+        // @ts-ignore
+        const cookie = jwt.encode({"exp": Date.now()/1000 + 1800}, config.captcha.secret_key + r.remoteAddress)
```
Comment (LaurenceJJones) [inline src/crowdsec.ts]: ~JWT are not encrypted so you are potentially leaking the captcha secret key here. Put an example key in https://jwt.io/ you can see its not encrypted~

Comment (LaurenceJJones) [reply src/crowdsec.ts:171]: You can site_key instead of secret key as that is client facing code

Comment (LaurenceJJones) [reply src/crowdsec.ts]: Actually I misread the code, you using that as the signing key

File: src/crowdsec.ts [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+        r.headersOut["Set-Cookie"] = ["crowdsec_captcha=" + cookie + "; Path=/; HttpOnly; SameSite=Strict"]
+        r.return(200, "OK")
+    } else{
+        r.return(401, "Unauthorized")
```
Comment (LaurenceJJones) [inline src/crowdsec.ts:175]: Should we respond with the captcha again instead of unauthorised?

**Comment by LaurenceJJones** [comment]
Just a final comment we can define multiple js_imports, so instead of bundling all npm modules into the file, we should create namespaces

For each module, @sbs2001 you said you had to make some modifications to the JS don't support you can document what changes you needed to make?

**Comment by sbs2001** [comment]
@LaurenceJJones the js_import imports  makes the module referenceable from the nginx config not the js module itself. njs doesn't have any package management system, the js_imports must be self contained.

**Comment by jaysee** [comment]
Hello, just to stay in touch, I will be happy when this will be merged/released!!

**Pull request merged**